src/js/utils/selection-utils.js

```diff
@@ -11,7 +11,23 @@ function comparePosition(selection) {
 
   const position = anchorNode.compareDocumentPosition(focusNode);
 
-  if (position & Node.DOCUMENT_POSITION_FOLLOWING) {
+  if (position & Node.DOCUMENT_POSITION_CONTAINS) {
+    return comparePosition({
+      focusNode: focusNode.childNodes[focusOffset],
+      focusOffset: 0,
+      anchorNode, anchorOffset
+    });
+  } else if (position & Node.DOCUMENT_POSITION_CONTAINED_BY) {
+    let offset = anchorOffset - 1;
+    if (offset < 0) {
+      offset = 0;
+    }
```

Comment:
IE picks DOM nodes higher up the tree than Chrome and Safari do. For example in `<div><span>abc</span><span>def</span></div>` with a cursor between `c` and `d` they might say the `focusNode` is `<div>` with an offset of `1`. However the `anchorNode` for a selection might still be `<span>` `2` if there was a selection. We make the upper element descend until a comparison of position can be made.

I can document, there are some test cases added.